test/micro/org/openjdk/bench/java/lang/foreign/CallOverheadByValue.java

```diff
@@ -0,0 +1,113 @@
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OperationsPerInvocation;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.annotations.Warmup;
+
+import java.lang.foreign.Arena;
+import java.lang.foreign.FunctionDescriptor;
+import java.lang.foreign.Linker;
+import java.lang.foreign.MemoryLayout;
+import java.lang.foreign.MemorySegment;
+import java.lang.foreign.SegmentAllocator;
+import java.lang.foreign.SymbolLookup;
+import java.lang.foreign.ValueLayout;
+import java.lang.invoke.MethodHandle;
+import java.util.concurrent.TimeUnit;
+
+import static org.openjdk.bench.java.lang.foreign.CLayouts.C_DOUBLE;
+
+@BenchmarkMode(Mode.AverageTime)
+@Warmup(iterations = 5, time = 500, timeUnit = TimeUnit.MILLISECONDS)
+@Measurement(iterations = 10, time = 500, timeUnit = TimeUnit.MILLISECONDS)
+@State(org.openjdk.jmh.annotations.Scope.Thread)
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+@Fork(value = 1, jvmArgs = {"-Xlog:gc", "--enable-native-access=ALL-UNNAMED", "-Djava.library.path=micro/native"})
+public class CallOverheadByValue {
+
+    public static final MemoryLayout POINT_LAYOUT = MemoryLayout.structLayout(
+            C_DOUBLE, C_DOUBLE
+    );
+    private static final MethodHandle MH_UNIT_ROTATED_BY_VALUE;
+    private static final MethodHandle MH_UNIT_ROTATED_BY_PTR;
+
+    static {
+        Linker abi = Linker.nativeLinker();
+        System.loadLibrary("CallOverheadByValue");
+        SymbolLookup loaderLibs = SymbolLookup.loaderLookup();
+        MH_UNIT_ROTATED_BY_VALUE = abi.downcallHandle(
+                loaderLibs.findOrThrow("unit_rotated"),
+                FunctionDescriptor.of(POINT_LAYOUT, C_DOUBLE)
+        );
+        MH_UNIT_ROTATED_BY_PTR = abi.downcallHandle(
+                loaderLibs.findOrThrow("unit_rotated_ptr"),
+                FunctionDescriptor.ofVoid(ValueLayout.ADDRESS, C_DOUBLE)
+        );
+    }
+
+    static final int NUM_CIRCLE_POINTS = 100;
+
+    Arena arena = Arena.ofConfined();
+    MemorySegment points = arena.allocate(POINT_LAYOUT, NUM_CIRCLE_POINTS);
+
+    @TearDown
+    public void tearDown() {
+        arena.close();
+    }
+
+    @Benchmark
+    @OperationsPerInvocation(NUM_CIRCLE_POINTS)
+    public MemorySegment byValue() throws Throwable {
+        for (int i = 0; i < NUM_CIRCLE_POINTS; i++) {
+            double phi = 2 * Math.PI * i / NUM_CIRCLE_POINTS;
+            // points[i] = unit_rotated(phi);
+            MemorySegment dest = points.asSlice(i * POINT_LAYOUT.byteSize(), POINT_LAYOUT.byteSize());
+            MemorySegment unused =
+                    (MemorySegment) MH_UNIT_ROTATED_BY_VALUE.invokeExact(
+                            (SegmentAllocator) (_, _) -> dest,
+                            phi);
+        }
```

Comment:
Would it be viable to measure just a single invocation? That way we can remove other things like the `asSlice` call, which might introduce noise.

Reply:
Done. Numbers are now:

```
Benchmark                    Mode  Cnt  Score   Error  Units
CallOverheadByValue.byPtr    avgt   10  3.291 ? 0.031  ns/op
CallOverheadByValue.byValue  avgt   10  5.464 ? 0.007  ns/op
```